fix: skipped tasks due to wrong when condition

a-belhadj opened the pull request:
# <!-- Intentionally left blank. -->

## Summary of Pull Request

base_additional_packages are not installed when building a Rocky Linux because it doesnt validate the first condition (distribution == Fedora)

Ansible doc : You can use logical operators to combine conditions. When you have multiple conditions that all need to be true (that is, a logical and), you can specify them as a list.(https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_conditionals.html)


In version 0.20, a similar issue occurs, cloud-init installation is always skipped for Rocky
## Type of Pull Request


- [x] This is a bugfix. `type/bug`
- [ ] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

Closes #950

## Test and Documentation Coverage

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [x] Tests have been completed.
- [ ] Documentation has been added or updated.

## Breaking Changes?

<!--
    Please check the one that applies to this pull request using "[x]".
-->

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

<!--
    If this pull request contains a breaking change, please describe the impact and mitigation path.
-->

tenthirtyam commented:
@a-belhadj - contributions require that your commit is signed. Please see the CONTRIBUTING.md on how to do this.

a-belhadj committed:
fix: skipped tasks due to wrong when condition

Ansible doc : You can use logical operators to combine conditions. When you have multiple conditions that all need to be true (that is, a logical and), you can specify them as a list.(https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_conditionals.html)

base_additional_packages are not installed when building a Rocky Linux because it doesnt validate the first condition (distribution == Fedora)

In version 0.20, a similar issue occurs, cloud-init installation is always skipped for Rocky.

a-belhadj commented:
> @a-belhadj - contributions require that your commit is signed. Please see the CONTRIBUTING.md on how to do this.

@tenthirtyam Done
